pvlib/iotools/pvgis.py

```diff
@@ -174,3 +179,97 @@ def _parse_pvgis_tmy_basic(src):
         data['time(UTC)'], format='%Y%m%d:%H%M', utc=True)
     data = data.drop('time(UTC)', axis=1)
     return data, None, None, None
+
+
+def read_pvgis_tmy(filename, pvgis_format=None):
+    """
+    Read a file downloaded from PVGIS.
+
+    Parameters
+    ----------
+    filename : str, pathlib.Path, or file-like buffer
+        Name, path, or buffer of file downloaded from PVGIS.
+    pvgis_format : str, default None
+        Format of PVGIS file or buffer. Equivalent to the ``outputformat``
+        parameter in the PVGIS TMY API. If `filename` is a file and
+        `pvgis_format` is ``None`` then the file extension will be used to
+        determine the PVGIS format to parse. For PVGIS files from the API with
+        ``outputformat='basic'``, please set `pvgis_format` to ``'basic'``. If
+        `filename` is a buffer, then `pvgis_format` is required and must be in
+        ``['csv', 'epw', 'json', 'basic']``.
+
+    Returns
+    -------
+    data : pandas.DataFrame
+        the weather data
+    months_selected : list
+        TMY year for each month, ``None`` for basic and EPW
+    inputs : dict
+        the inputs, ``None`` for basic and EPW
+    meta : list or dict
+        meta data, ``None`` for basic
+
+    Raises
+    ------
+    ValueError
+        if `pvgis_format` is ``None`` and `filename` is either a buffer or the
+        file extension is neither ``.csv``, ``.json``, nor ``.epw``, or if
+        `pvgis_format` provided as input isn't in
+        ``['csv', 'epw', 'json', 'basic']``
+
+    See also
+    --------
+    get_pvgis_tmy
+    """
+    # get the PVGIS outputformat
+    if pvgis_format is None:
+        # get the file extension from everything after the last dot on the
+        # right and make sure it's lower case to compare with epw, csv, or json
+        outputformat = str(filename).lower().rsplit('.', 1)[-1]
```

Comment:
can we use pathlib's suffix? https://docs.python.org/3/library/pathlib.html#pathlib.PurePath.suffix

Reply:
interesting, only if we convert it to a pathlib object first, we could require a pathlib object, but that seems a bit restrictive. I wonder what happens if you try to make a pathlib object from a pathlib object, probably nothing. I'll try it and report back, ok?

Reply:
yep, that seems to work, is it your prefernce?
```python
In [2]: from pvlib.tests.conftest import DATA_DIR

In [3]: greensboro = DATA_DIR / '723170TYA.CSV'

In [4]: greensboro
Out[4]: WindowsPath('pvlib/data/723170TYA.CSV')

In [5]: import pathlib

In [6]: pathlib.Path(greensboro)
Out[6]: WindowsPath('pvlib/data/723170TYA.CSV')

In [7]: pathlib.Path(greensboro).suffix
Out[7]: '.CSV'
```